plugins/transcribe.py

```diff
@@ -0,0 +1,191 @@
+            'content-type': 'audio/raw;encoding=signed-integer;bits=16;rate=8000;endian=little',
+        }
+        try:
+            async with ClientSession() as session:
+                async with session.post(
+                        f"{self.api_url}/speech", headers=headers,
+                        data=io.BufferedReader(io.BytesIO(chunk.raw_data))) as resp:
+                    if resp.status == 200:
+                        response = await resp.json()
+                        text = response['_text'] if '_text' in response else response['text']
+        except Exception as e:
+            error = f"Could not transcribe chunk: {e}\n{traceback.format_exc()}"
+
+        return text, error
+
+    @staticmethod
+    async def __generate_chunks(segment, length=20000 / 1001):
+        """
+        Based on https://github.com/charslab/TranscriberBot/blob/
+        823b1423832b7117ad41c83abb3e25d58dd9e789/
+        src/audiotools/speech.py#L49
+        """
+        return [segment[i:i + int(length * 1000)]
+                for i in range(0, len(segment), int(length * 1000))]
+
+    @staticmethod
+    async def __preprocess_audio(audio):
+        """
+        From https://github.com/charslab/TranscriberBot/blob/
+        823b1423832b7117ad41c83abb3e25d58dd9e789/
+        src/audiotools/speech.py#L67
+        """
+        return audio.set_sample_width(2).set_channels(1).set_frame_rate(8000)
+
+    async def transcribe(self, path):
+        """
+        Based on https://github.com/charslab/TranscriberBot/blob/
+        823b1423832b7117ad41c83abb3e25d58dd9e789/
+        src/audiotools/speech.py#L70
+        """
+        logger.info("Transcribing file %s", path)
+        try:
+            audio = AudioSegment.from_file(path)
+            chunks = await self.__generate_chunks(await self.__preprocess_audio(audio))
+            self.chunks = len(chunks)
+            logger.info("Got %d chunks", len(chunks))
+
+            for i, chunk in enumerate(chunks):
+                logger.info("Transcribing chunk %d", i)
+                text, error = await self.__transcribe_chunk(chunk, self.lang)
+                self.text += text
+                yield text, error
+
+        except CouldntDecodeError:
+            yield None, "`Error decoding the audio file. " \
+                        "Ensure that the provided audio is a valid audio file!`"
+
+
+@userge.on_cmd("stt", about={
+    'header': "transcribe a file (speech to text)",
+    'options': {'-t': 'send text to telegram as well as the transcription file'},
+    'usage': "{tr}stt lang [file / folder path | direct link | reply to telegram file]",
+    'examples': ['{tr}stt en link', '{tr}stt ar -t link']}, check_downpath=True)
+async def stt_(message: Message):
+    """ Speech to text using Wit.ai """
+    send_text = bool('v' in message.flags)
+    replied = message.reply_to_message
+    message_id = replied.message_id if replied \
+        else message.reply_to_message.message_id or message.message_id
```

Comment:
both condition same 😢😢

it should be 

`replied.message_id if replied else message.message_id`

Reply:
@Krishna-Singhal updated.